doc/03_research/02_perception/first_implementation_plan.md

```diff
@@ -0,0 +1,157 @@
+- **Instance Segmentation**: \
+    Detection of the different instances of things.
+- **Panoptic Segmentation**: \
+    Combination of semantic segmentation and instance segmentation. Detection of stuff plus instances of things.
+
+![](../../00_assets/segmentation.png)
+[Source](https://www.v7labs.com/blog/panoptic-segmentation-guide)
+
+### Image Panoptic Segmentation
+
+Possible implementation with [EfficientPS](https://arxiv.org/pdf/2004.02307.pdf).
+This net has a pretty high panoptic quality of 64-67% and a relatively low calculation 
+time of 166ms on a 1024x2048 image.
+
+###  LIDAR Panoptic Segmentation
+
+**Discussion**: \
+This is probably necessary to get also an overview about entities that are not in the field of view of the cameras 
+like vehicles behind or next to the vehicle.
+
+An efficient way to compute LIDAR panoptic segmentation is [EfficientLPS](https://arxiv.org/pdf/2102.08009.pdf) ([Website](http://lidar-panoptic.cs.uni-freiburg.de/)).
+It has a panoptic quality between 53% and 59% and a calculation time of 154-213ms depending on the chosen architecture.
+
+---
+## Position Validation
+
+If some positioning sensors show a significant deviation, the task of this module is to 
+clear these deviations and to calculate the most accurate position possible.
+
+---
+## Obstacle Detection and Object Classification
+
+This module is the first with a direct output into the planning layer. \
+Its tasks are to combine the information given by the map, cameras, panoptic segmentation 
+and LIDAR (panoptic segmentation).
+
+A possible way for the combination of camera and LIDAR data is e.g. a projection between 
+the semantic image and LIDAR points to generate a panoptic semantic LIDAR point cloud.
+
+This enables to calculate bounding boxes for each thing and to get a precise overview about the surrounding stuff.
+
+---
+## Lane Detection
+
+For lane detection, all occurring lines are classified (stop line, solid middle line, dashed middle line, etc.)
+Based on this information, the lanes are classified (own lane, driveable, oncoming lane etc.). Information given by the
+map and GNSS are used either.
+No implementation plan yet.
+
+---
+## Traffic Light Detection 
+
+This module is responsible for the detection of traffic lights and the classification of their current state. \
+It combines the information about position given by the map and Obstacle Detection and Object Classification Module and 
+the information about the state given by the camera images.
+
+For state classification, the part showing the traffic light is snipped out of the camera image to get a better and 
+faster prediction due to a smaller image size and less information. To prevent the classification network to classify 
+the state always as orange on traffic lights with orange housings, the housing is masked out. 
+As classification net I would recommend the net implemented by PAF21-1.
```

Comment:
Is there a link to the according code or even documentation to this implementation?

Reply:
Solved in [cf3092b](https://github.com/ll7/paf22/pull/59/commits/cf3092bacaf21e954fab86c12c076c9a1044ae03)